programs/drift/src/instructions/user.rs

```diff
@@ -1896,6 +1902,517 @@ pub fn handle_transfer_perp_position<'c: 'info, 'info>(
+        ErrorCode::MarketBeingInitialized,
+        "Market is being initialized"
+    )?;
+
+    controller::spot_balance::update_spot_market_cumulative_interest(
+        &mut spot_market,
+        Some(&oracle_price_data),
+        now,
+    )?;
+
+    user.increment_total_deposits(
+        amount,
+        oracle_price_data.price,
+        spot_market.get_precision().cast()?,
+    )?;
+
+    let total_deposits_after = user.total_deposits;
+    let total_withdraws_after = user.total_withdraws;
+
+    {
+        let perp_position = user.force_get_isolated_perp_position_mut(perp_market_index)?;
+
+        update_spot_balances(
+            amount.cast::<u128>()?,
+            &SpotBalanceType::Deposit,
+            &mut spot_market,
+            perp_position,
+            false,
+        )?;
+    }
+
+    validate!(
+        matches!(spot_market.status, MarketStatus::Active),
+        ErrorCode::MarketActionPaused,
+        "spot_market not active",
+    )?;
+
+    drop(spot_market);
+
+    if user.is_isolated_position_being_liquidated(perp_market_index)? {
+        // try to update liquidation status if user is was already being liq'd
+        let is_being_liquidated = is_isolated_position_being_liquidated(
+            user,
+            &perp_market_map,
+            &spot_market_map,
+            &mut oracle_map,
+            perp_market_index,
+            state.liquidation_margin_buffer_ratio,
+        )?;
+
+        if !is_being_liquidated {
+            user.exit_isolated_position_liquidation(perp_market_index)?;
+        }
+    }
+
+    user.update_last_active_slot(slot);
+
+    let spot_market = &mut spot_market_map.get_ref_mut(&spot_market_index)?;
+
+    controller::token::receive(
+        &ctx.accounts.token_program,
+        &ctx.accounts.user_token_account,
+        &ctx.accounts.spot_market_vault,
+        &ctx.accounts.authority,
+        amount,
+        &mint,
+        if spot_market.has_transfer_hook() {
+            Some(remaining_accounts_iter)
+        } else {
+            None
+        },
+    )?;
+    ctx.accounts.spot_market_vault.reload()?;
+
+    let deposit_record_id = get_then_update_id!(spot_market, next_deposit_record_id);
+    let oracle_price = oracle_price_data.price;
+
+    let deposit_record = DepositRecord {
+        ts: now,
+        deposit_record_id,
+        user_authority: user.authority,
+        user: user_key,
+        direction: DepositDirection::Deposit,
+        amount,
+        oracle_price,
+        market_deposit_balance: spot_market.deposit_balance,
+        market_withdraw_balance: spot_market.borrow_balance,
+        market_cumulative_deposit_interest: spot_market.cumulative_deposit_interest,
+        market_cumulative_borrow_interest: spot_market.cumulative_borrow_interest,
+        total_deposits_after,
+        total_withdraws_after,
+        market_index: spot_market_index,
+        explanation: DepositExplanation::None,
+        transfer_user: None,
```

Comment:
are we planning on tracking isolated positions deposit/withdraws? might need something here

Reply:
yeah good question for lukas/kev